### PR TITLE
[lc_ctrl] Remove RD access to isolated partition in DEV

### PR DESCRIPTION
**Pull request by msfschaffner**
This addresses a security concern in #7197.

Signed-off-by: Michael Schaffner <msf@opentitan.org>

### CODE REVIEW

**Comment by msfschaffner** [comment]
We can also remove the write access, but I don't think that is needed.

**Review by tjaychen — APPROVED** [review]
i think this is what we want, but can we hold off on merging until we get a chance to talk it through with the software team?

**Comment by msfschaffner** [comment]
Sure - just thought I'd put the change already out such that we can get it reviewed.

**Review by cindychip — APPROVED** [review]

File: hw/ip/lc_ctrl/rtl/lc_ctrl_signal_decode.sv
Comment (zi-v) [inline hw/ip/lc_ctrl/rtl/lc_ctrl_signal_decode.sv:132]: Consider changing this comment to reflect the new restriction

**Review by zi-v — APPROVED** [review]
LGTM

File: hw/ip/lc_ctrl/rtl/lc_ctrl_signal_decode.sv
Comment (msfschaffner) [reply hw/ip/lc_ctrl/rtl/lc_ctrl_signal_decode.sv]: Thanks, I have expanded this comment a bit.

**Pull request merged**
